src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -257,7 +282,10 @@ void OdometryPublisherPrivate::UpdateOdometry(
     return;
 
   // Get and set robotBaseFrame to odom transformation.
-  const math::Pose3d pose = worldPose(this->model.Entity(), _ecm);
+  const math::Pose3d rawPose = worldPose(this->model.Entity(), _ecm);
+  math::Pose3d pose;
+  pose.Pos() = rawPose.Pos() + this->offset.Pos();
+  pose.Rot() = this->offset.Rot() * rawPose.Rot();
```

Comment:
I see that these two lines are copying the logic from the old p3d plugin, but I'm trying to figure out what in which frame they are expressed. I think they must be in the world frame and not the body-fixed frame? @azeey can you confirm?

Reply:
Yup, seems to be the pose in the world frame as per the implementation in ``Util.cc`` : https://github.com/ignitionrobotics/ign-gazebo/blob/c08fc87e2b965ab388f6dc4a74f8668213ecae8f/src/Util.cc#L74

Reply:
The way it is currently implemented, which is exactly like the [old p3d plugin](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/231a7219b36b8a6cdd100b59f66a3df2955df787/gazebo_plugins/src/gazebo_ros_p3d.cpp#L301-L303), the offset doesn't appear to be in the world frame nor in the body-fixed frame. The code in the p3d plugin might have been written that way because there was an error in the offset generation in libsdformat till recently (see https://github.com/ignitionrobotics/sdformat/pull/500).

If the offset is in world frame, I think the code should be:
```
  pose.Pos() = this->offset.Rot() * rawPose.Pos() + this->offset.Pos();
  pose.Rot() = this->offset.Rot() * rawPose.Rot();

  // Or simply:
  pose = this->offset * rawPose;
```

and if the offset is in body-fixed frame, it should be

```
  pose.Pos() = rawPose.Rot() * this->offset.Pos() + rawPose.Pos();
  pose.Rot() = rawPose.Rot() * this->offset.Rot() ;

  // Or simply:
  pose = rawPose * this->offset;

```

And from this: https://github.com/ignitionrobotics/sdformat/pull/500#issuecomment-785302490, I think the it should be in the body-fixed frame.

Reply:
Done !